sdk/core/azure-core-http-okhttp/src/test/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilderTests.java

```diff
@@ -261,32 +282,94 @@ public void nullDispatcherThrows() {
+     * Tests that building a client with a proxy will send the request through the proxy server.
      */
     @ParameterizedTest
-    @EnumSource(ProxyOptions.Type.class)
-    public void buildWithProxy(ProxyOptions.Type proxyType) {
-        String expectedProxyHost = "localhost";
-        int expectedProxyPort = 8888;
-
-        ProxyOptions proxyOptions = new ProxyOptions(proxyType,
-            new InetSocketAddress(expectedProxyHost, expectedProxyPort)).setCredentials("1", "1");
-
-        OkHttpClient validatorClient = new OkHttpClient.Builder()
-            .eventListener(new TestEventListenerValidator(expectedProxyHost, expectedProxyPort))
-            .build();
+    @MethodSource("buildWithProxySupplier")
+    public void buildWithProxy(boolean shouldHaveProxy, Proxy.Type proxyType, ProxyOptions proxyOptions,
+        String requestUrl) {
+        OkHttpClient validatorClient = okHttpClientWithProxyValidation(shouldHaveProxy, proxyType);
 
-        OkHttpAsyncHttpClient okClient = (OkHttpAsyncHttpClient) new OkHttpAsyncHttpClientBuilder(validatorClient)
+        HttpClient okClient = new OkHttpAsyncHttpClientBuilder(validatorClient)
             .proxy(proxyOptions)
             .build();
 
-        StepVerifier.create(okClient.send(new HttpRequest(HttpMethod.GET, defaultUrl)))
-            .verifyError();
+        StepVerifier.create(okClient.send(new HttpRequest(HttpMethod.GET, requestUrl)))
+            .verifyErrorMatches(throwable -> throwable.getMessage()
+                .contains(TestEventListenerValidator.EXPECTED_EXCEPTION_MESSAGE));
+    }
+
+    private static Stream<Arguments> buildWithProxySupplier() {
+        InetSocketAddress proxyAddress = new InetSocketAddress("localhost", 12345);
+
+        ProxyOptions socks4Proxy = new ProxyOptions(ProxyOptions.Type.SOCKS4, proxyAddress);
+        ProxyOptions socks5Proxy = new ProxyOptions(ProxyOptions.Type.SOCKS5, proxyAddress);
+        ProxyOptions simpleHttpProxy = new ProxyOptions(ProxyOptions.Type.HTTP, proxyAddress);
+
+        List<Arguments> arguments = new ArrayList<>();
+
+        /*
+         * Simple non-authenticated proxies without non-proxy hosts configured.
+         */
+        arguments.add(Arguments.of(true, Proxy.Type.SOCKS, socks4Proxy, defaultUrl));
+        arguments.add(Arguments.of(true, Proxy.Type.SOCKS, socks5Proxy, defaultUrl));
+        arguments.add(Arguments.of(true, Proxy.Type.HTTP, simpleHttpProxy, defaultUrl));
+
+        /*
+         * HTTP proxy with authentication configured.
+         */
+        ProxyOptions authenticatedHttpProxy = new ProxyOptions(ProxyOptions.Type.HTTP, proxyAddress)
+            .setCredentials("1", "1");
+
+        arguments.add(Arguments.of(true, Proxy.Type.HTTP, authenticatedHttpProxy, defaultUrl));
+
+        /*
+         * Information for non-proxy hosts testing.
+         */
+        String rawNonProxyHosts = String.join("|", "localhost", "127.0.0.1", "*.microsoft.com", "*.linkedin.com");
```

Comment:
Given that we are now using `find()` instead of `matches()`, can "*.microsoft.com" just be ".microsoft.com"? The `*` prefix may not be necessary now if that was the intention to switch to `find()`.